rpc/src/abci.rs

```diff
@@ -0,0 +1,34 @@
+//! Old ABCI structures, formerly defined in `tendermint::abci`.
```

Comment:
Yep, that's the eventual goal:

> I decoupled the original ABCI-for-RPC types from the new ABCI-for-ABCI types by copying them to the RPC crate. While this means the RPC crate now has duplicate definitions of ABCI types, it means that that code can be refactored with the ABCI code in chunks, rather than getting it all done in this PR.